### PR TITLE
mac: replace sprintf uses with snprintf.

### PR DESCRIPTION
**Pull request by sezero**
Reference issue: https://github.com/libusb/hidapi/pull/509

### CODE REVIEW

**Review by Youw — APPROVED** [review]
Thanks

File: mac/hid.c [hunk 1]
```diff
@@ -513,12 +513,7 @@ static struct hid_device_info *create_device_info_with_usage(IOHIDDeviceRef dev,
 		   9+1+20+1=31 bytes buffer, but allocate 32 for simple alignment */
 		cur_dev->path = calloc(1, 32);
 		if (cur_dev->path != NULL) {
-			/* Yes, compiler, we know that snprintf is preferable,
-			   but we're not ready to abandon older macOS-es/SDKs where it is not yet available */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-			sprintf(cur_dev->path, "DevSrvsID:%llu", entry_id);
-#pragma GCC diagnostic pop
+			snprintf(cur_dev->path, 32, "DevSrvsID:%llu", entry_id);
```
Comment (Youw) [inline mac/hid.c:516]: I'd prefer if we had a local named constant for the `32` value

File: mac/hid.c [hunk 2]
```diff
@@ -988,16 +983,7 @@ hid_device * HID_API_EXPORT hid_open_path(const char *path)
 	dev->max_input_report_len = (CFIndex) get_max_report_length(dev->device_handle);
 	dev->input_report_buf = (uint8_t*) calloc(dev->max_input_report_len, sizeof(uint8_t));
 
-
-/* Yes, compiler, we know that snprintf is preferable,
-   but we're not ready to abandon older macOS-es/SDKs where it is not yet available */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-	/* Create the Run Loop Mode for this device.
-	   printing the reference seems to work. */
```
[the remark this answers is not in the document]
Comment (sezero) [reply mac/hid.c]: Oops - was an accident, restored now.

File: mac/hid.c (hunk 1 above)
Comment (sezero) [reply mac/hid.c]: I had followed what you already had. Invented a `const size_t path_len` for it now.

Comment (Youw) [reply mac/hid.c]: true, I tend to be lazy about named constants, but usually not when it comes to multiple usages of such

**Pull request merged**